jetty-websocket/javax-websocket-server/src/main/java/org/eclipse/jetty/websocket/javax/server/JavaxWebSocketServerContainer.java

```diff
@@ -71,34 +80,86 @@ public static WebSocketContainer getWebSocketContainer()
+    {
+        ContextHandler contextHandler = ContextHandler.getContextHandler(servletContext);
+        JavaxWebSocketServerContainer container = contextHandler.getBean(JavaxWebSocketServerContainer.class);
+        if (container==null)
+        {
+            // Find Pre-Existing (Shared?) HttpClient and/or executor
+            HttpClient httpClient = (HttpClient)servletContext.getAttribute(JavaxWebSocketServerContainerInitializer.HTTPCLIENT_ATTRIBUTE);
+            if (httpClient == null)
+                httpClient = (HttpClient)contextHandler.getServer()
+                    .getAttribute(JavaxWebSocketServerContainerInitializer.HTTPCLIENT_ATTRIBUTE);
+
+            Executor executor = httpClient == null?null:httpClient.getExecutor();
+            if (executor == null)
+                executor = (Executor)servletContext
+                    .getAttribute("org.eclipse.jetty.server.Executor");
+            if (executor == null)
+                executor = contextHandler.getServer().getThreadPool();
+
+            if (httpClient != null && httpClient.getExecutor() == null)
+                httpClient.setExecutor(executor);
+
+            // Create the Jetty ServerContainer implementation
+            container = new JavaxWebSocketServerContainer(
+                WebSocketMapping.ensureMapping(servletContext), httpClient, executor);
+            contextHandler.addManaged(container);
+            contextHandler.addLifeCycleListener(container);
+        }
+        // Store a reference to the ServerContainer per - javax.websocket spec 1.0 final - section 6.4: Programmatic Server Deployment
+        servletContext.setAttribute(ServerContainer.class.getName(), container);
+        return container;
     }
 
-    private final WebSocketCreatorMapping _webSocketCreatorMapping;
+    private final WebSocketMapping _webSocketMapping;
     private final JavaxWebSocketServerFrameHandlerFactory frameHandlerFactory;
     private final Executor executor;
+    private final FrameHandler.ConfigurationCustomizer customizer = new FrameHandler.ConfigurationCustomizer();
     private long asyncSendTimeout = -1;
     private List<Class<?>> deferredEndpointClasses;
     private List<ServerEndpointConfig> deferredEndpointConfigs;
 
     /**
      * Main entry point for {@link JavaxWebSocketServerContainerInitializer}.
-     *
-     * @param webSocketCreatorMapping the {@link WebSocketCreatorMapping} that this container belongs to
+     * @param webSocketMapping the {@link WebSocketMapping} that this container belongs to
      * @param httpClient              the {@link HttpClient} instance to use
      */
-    public JavaxWebSocketServerContainer(WebSocketCreatorMapping webSocketCreatorMapping, HttpClient httpClient, Executor executor)
+    public JavaxWebSocketServerContainer(WebSocketMapping webSocketMapping, HttpClient httpClient, Executor executor)
```

Comment:
This is JSR356, this MUST be a `JsrCreator` based mapping then.

Reply:
I don't understand your comment here?  the mapping still contains creators, but it also has factories and customizers, so it maps a tuple not just a creator - so the name has been changed to not just be about creators.